src/_utils/_git_secret_tools_osx.sh

```diff
@@ -13,3 +13,8 @@ function __temp_file_osx {
   filename=$(mktemp -t _gitsecrets_XXX )
   echo "$filename";
 }
+
+
+function __sha256_osx {
+  /usr/bin/shasum -a256 "$1"
```

Comment:
Do we really need `/usr/bin` part here?

Reply:
It's somewhat safer because the path to the system `shasum` is not in doubt on macOS, so any other random `shasum` would almost certainly be undesirable.

Reply:
```
Josephs-MacBook-Pro:~ joe$ mdutil -t `which shasum`
/usr/local/Cellar/perl/5.26.1/bin/shasum
Josephs-MacBook-Pro:~ joe$ 
```
That is definitely not what you'd want to be using.

Reply:
To be clear `mdutil -t` is just a cheat for `realpath`, which macOS doesn't have.
```
Josephs-MacBook-Pro:~ joe$ which shasum
/usr/local/bin/shasum
```

Reply:
What's the difference between them? I am concerned about it, because we don't have any other absolute paths.

Reply:
The /usr/local/bin/shasum is provided by Homebrew's perl, not provided by the system, and is subject to breakage if the user's perl installation happens to be messed up. See https://github.com/Homebrew/brew/issues/3281 for the nightmares it can bring to your life.

Reply:
We ended up using the absolute path as of https://github.com/Homebrew/brew/pull/3330 because random issues with non-system `shasum` kept coming up.

Reply:
Ok, it makes sense. Thanks!

Reply:
You're welcome!